internal/handlers/handler_firstfactor.go

```diff
@@ -36,15 +38,14 @@ func FirstFactorPost(ctx *middlewares.AutheliaCtx) {
 	if err != nil {
 		ctx.Logger.Debugf("Mark authentication attempt made by user %s", bodyJSON.Username)
 		ctx.Providers.Regulator.Mark(bodyJSON.Username, false) //nolint:errcheck // TODO: Legacy code, consider refactoring time permitting.
-
-		ctx.Error(fmt.Errorf("Error while checking password for user %s: %s", bodyJSON.Username, err.Error()), authenticationFailedMessage)
+		handleErrorResponse(ctx, fmt.Errorf("Error while checking password for user %s: %s", bodyJSON.Username, err.Error()), authenticationFailedMessage)
 		return
 	}
 
 	if !userPasswordOk {
 		ctx.Logger.Debugf("Mark authentication attempt made by user %s", bodyJSON.Username)
 		ctx.Providers.Regulator.Mark(bodyJSON.Username, false) //nolint:errcheck // TODO: Legacy code, consider refactoring time permitting.
-
+		ctx.SetStatusCode(fasthttp.StatusUnauthorized)
```

Comment:
you should also use `handleErrorResponse` here otherwise there is an easy way to know that branch has been reached by a hacker. I did it wrong at the beginning, we don't want to differentiate the message.